extensions/kafka/src/main/java/io/deephaven/kafka/ingest/KafkaIngester.java

```diff
@@ -355,9 +360,29 @@ private void consumerLoop() {
             }
             final long beforePoll = System.nanoTime();
             final long remainingNanos = beforePoll > nextReport ? 0 : (nextReport - beforePoll);
-            consumerLoopCallbacks.forEach(x -> x.beforePoll(kafkaConsumer));
-            final boolean more = pollOnce(Duration.ofNanos(remainingNanos));
-            consumerLoopCallbacks.forEach(x -> x.afterPoll(kafkaConsumer, more));
+
+            boolean more = true;
+            if (consumerLoopCallback != null) {
+                try {
+                    consumerLoopCallback.beforePoll(kafkaConsumer);
+                } catch (Exception e) {
+                    log.error().append(logPrefix).append("Exception while executing beforePoll callback:").append(e)
+                            .append(", aborting.").endl();
+                    more = false;
+                }
+            }
+            if (more) {
+                more = pollOnce(Duration.ofNanos(remainingNanos));
+                if (consumerLoopCallback != null) {
+                    try {
+                        consumerLoopCallback.afterPoll(kafkaConsumer, more);
+                    } catch (Exception e) {
+                        log.error().append(logPrefix).append("Exception while executing afterPoll callback:").append(e)
+                                .append(", aborting.").endl();
+                        more = false;
+                    }
+                }
+            }
```

Comment:
We need to deliver errors to the consumers...

Reply:
I followed the existing pattern in pollOnce where the poll() method could return an error:
        final ConsumerRecords<?, ?> records;
        try {
            ++pollCalls;
            records = kafkaConsumer.poll(timeout);
        } catch (WakeupException we) {
            // we interpret a wakeup as a signal to stop /this/ poll.
            return true;
        } catch (Exception ex) {
            log.error().append(logPrefix).append("Exception while polling for Kafka messages:").append(ex)
                    .append(", aborting.").endl();
            return false;
        }

Reply:
We discussed offline that I fixed that bug yesterday, and you have generalized a solution from that fix. Marking resolved.